manufacturing_order_type/models/mrp_production.py

```diff
@@ -41,3 +41,13 @@ def create(self, vals):
             if mo_type.sequence_id:
                 vals["name"] = mo_type.sequence_id.next_by_id()
         return super().create(vals)
+
+    @api.constrains('company_id', 'mo_type_id.company_id')
+    def _constrains_company(self):
+        if self.company_id not in self.mo_type_id.company_id:
+            raise exceptions.ValidationError("Error : Company and Type No Matching !!!")
```

Comment:
better import as
`from odoo.exceptions import ValidationError`
and then use (and make the generic message, no screaming !!! :)
```suggestion
            raise ValidationError(_("Document's company and type's company mismatch")
```